lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,380 @@
+          selectedUserIds: selectedUserIds),
+        Expanded(
+          child: _NewDmUserList(
+            filteredUsers: filteredUsers,
+            selectedUserIds: selectedUserIds,
+            onUserTapped: (userId) => _handleUserTap(userId))),
+      ]);
+  }
+}
+
+class _NewDmHeader extends StatelessWidget {
+  const _NewDmHeader({required this.selectedUserIds});
+
+  final Set<int> selectedUserIds;
+
+  Widget _buildBackButton(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    return GestureDetector(
+      onTap: Navigator.of(context).pop,
+      child: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Icon(ZulipIcons.chevron_left, size: 24, color: designVariables.icon),
+          Text(zulipLocalizations.newDmSheetBackButtonLabel,
+            style: TextStyle(
+              color: designVariables.icon,
+              fontSize: 20,
+              height: 30 / 20)),
+        ]));
+  }
+
+  Widget _buildNextButton(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    final nextButtonColor = selectedUserIds.isEmpty
+      ? designVariables.icon.withFadedAlpha(0.5)
+      : designVariables.icon;
+
+    return GestureDetector(
+      onTap: selectedUserIds.isEmpty ? null : () {
+        final narrow = DmNarrow.withUsers(
+          selectedUserIds.toList(),
+          selfUserId: store.selfUserId);
+        Navigator.pop(context);
+        Navigator.push(context,
+          MessageListPage.buildRoute(context: context, narrow: narrow));
+      },
+      child: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Text(zulipLocalizations.newDmSheetNextButtonLabel,
+            style: TextStyle(
+              color: nextButtonColor,
+              fontSize: 20,
+              height: 30 / 20,
+            ).merge(weightVariableTextStyle(context, wght: 600))),
+          Icon(ZulipIcons.chevron_right,
+            size: 24,
+            color: nextButtonColor),
+        ]));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    return Container(
+      padding: const EdgeInsetsDirectional.fromSTEB(8, 10, 8, 6),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          _buildBackButton(context),
+          Expanded(
+            child: Text(zulipLocalizations.newDmSheetScreenTitle,
+              style: TextStyle(
+                color: designVariables.title,
+                fontSize: 20,
+                height: 30 / 20,
+              ).merge(weightVariableTextStyle(context, wght: 600)),
+              overflow: TextOverflow.ellipsis,
```

Comment:
Does this `overflow` parameter have any effect without `maxLines`? Reading docs, I believe it doesn't.